raft/server.cc

```diff
@@ -495,13 +510,14 @@ future<add_entry_reply> server_impl::execute_add_entry(server_id from, command c
     if (from != _id && !_fsm->get_configuration().contains(from)) {
         // Do not accept entries from servers removed from the
         // configuration.
-        co_return add_entry_reply{not_a_leader{server_id{}}};
+        co_return add_entry_reply{transient_error{format("Add entry from {} was discarded since "
+                                                         "it is not part of the configuration", from), false, {}}};
     }
     logger.trace("[{}] adding a forwarded entry from {}", id(), from);
     try {
         co_return add_entry_reply{co_await add_entry_on_leader(std::move(cmd), as)};
     } catch (raft::not_a_leader& e) {
-        co_return add_entry_reply{e};
+        co_return add_entry_reply{transient_error{std::current_exception(), false, e.leader}};
```

Comment:
Why no delay?

Reply:
Is there a guarantee that `e.leader` is not `{}` here?

Reply:
> There is a possibility that the leader here is not a leader any more, but a follower still thinks that it is the leader and it will retry immediately, but soon enough it will learn about new leader.

This is not good enough. For me, the main point was getting rid of the tight retry loops like the one in https://github.com/scylladb/scylladb/issues/11564. In the situation you describe we could have a loop like that for a few seconds.

Reply:
The `delay` just makes the entire thing full of such "cases" and "possibilities". That's why I think it would be better to just remove it. It doesn't really matter if we wait these 100ms, but it would simplify our reasoning greatly.